### PR TITLE
A couple of bugfixes in handling of first arg after "("

### LINKED ISSUE #1703: scalafmt crashes with head of empty list error on anonymous SAM
This template is a guideline, not a strict requirement.
- **Configuration**:
```
version = "2.4.0-RC2"
maxColumn = 120 
rewrite.rules = [PreferCurlyFors, SortModifiers, AvoidInfix, SortImports]
trailingCommas = always
assumeStandardLibraryStripMargin = true
optIn.breakChainOnFirstMethodDot = false
spaces.inImportCurlyBraces = true
includeCurlyBraceInSelectChains = true
spaces.beforeContextBoundColon = Always
// These are copied from  https://scalameta.org/scalafmt/docs/configuration.html#other
rewrite.neverInfix.excludeFilters = [
  until
  to
  by
  eq
  ne
  "should.*"
  "contain.*"
  "must.*"
  in
  ignore
  be
  taggedAs
  thrownBy
  synchronized
  have
  when
  size
  only
  noneOf
  oneElementOf
  noElementsOf
  atLeastOneElementOf
  atMostOneElementOf
  allElementsOf
  inOrderElementsOf
  theSameElementsAs
  // these are new
  throws
  returns
  satisfy
]


```

## Steps

Given code like this:
```scala
object Foo {
  def foo(): Unit = ()
  
  def run(): Thread = {
    new Thread(() => foo())
  }
}
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt formats produces the error 
```
head of empty list: [...]/Foo.scala
```

## Workaround

This formats fine: 

```scala
object Foo {
  def foo(): Unit = ()

  def run(): Thread = {
    val r: Runnable = () => foo()
    new Thread(r)
  }
}
```

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #1703 

`scala-repos` diffs:
* Router bugfix: split, indent comments consistently https://github.com/kitbellew/scala-repos/commit/93a939860e153783b1e3e6d79005d0ef2381a386
* TreeOps bugfix: find Init param group correctly https://github.com/kitbellew/scala-repos/commit/68cb5977aadcfc68466ebfff49e4e81e966f4987?w=1
* Router bugfix: consistently allow break on "(/**/" https://github.com/kitbellew/scala-repos/commit/3fd4a1a7241282b43aed2615b5383bc0f4588511?w=1

### CODE REVIEW

**Comment by kitbellew** [comment]
@olafurpg @tanishiking @poslegm PTAL

**Comment by kitbellew** [comment]
@olafurpg @tanishiking @poslegm would appreciate your insights. i would prefer not to release v2.4.0 without these fixes (follow-on to changes included in this release). thanks!

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatToken.scala [hunk 1]
```diff
@@ -43,7 +44,9 @@ object FormatToken {
     */
   case class Meta(
       between: Array[Token],
-      idx: Int
+      idx: Int,
+      lowner: Tree,
+      rowner: Tree
```
Comment (tanishiking) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatToken.scala]: What about `leftOwner`, `rightOwner` instead?
Personally, I prefer explicit names in Scala.

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatToken.scala]: no problem, will do.

**Pull request merged**